src/commands/openInRemoteSsh.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import { join } from 'path';
+import * as SSHConfig from 'ssh-config';
+import { commands } from 'vscode';
+import { IActionContext } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { VirtualMachineTreeItem } from '../tree/VirtualMachineTreeItem';
+import { sshFsPath } from '../utils/sshUtils';
+import { addSshKey } from './addSshKey';
+import { verifyRemoteSshExtension } from './verifyRemoteSshExtension';
+
+export async function openInRemoteSsh(context: IActionContext, node?: VirtualMachineTreeItem): Promise<void> {
+    if (!node) {
+        node = await ext.tree.showTreeItemPicker<VirtualMachineTreeItem>(VirtualMachineTreeItem.linuxContextValue, context);
+    }
+
+    await verifyRemoteSshExtension(context);
+
+    const sshConfigPath: string = join(sshFsPath, 'config');
+    await fse.ensureFile(sshConfigPath);
+    const configFile: string = (await fse.readFile(sshConfigPath)).toString();
+    const sshConfig: SSHConfig.HostConfigurationDirective[] = <SSHConfig.HostConfigurationDirective[]><unknown>SSHConfig.parse(configFile);
+    const hostName: string = await node.getIpAddress();
+    let host: string = node.name;
+    let foundHostName: boolean = false;
+
+    for (const hostEntry of sshConfig) {
+        for (const config of hostEntry.config) {
+            const castedConfig: SSHConfig.BaseConfigurationDirective = <SSHConfig.BaseConfigurationDirective>config;
+            if (castedConfig.param === 'HostName' && castedConfig.value === hostName) {
+                host = Array.isArray(hostEntry.value) ? hostEntry.value[0] : hostEntry.value;
+                foundHostName = true;
+                break;
+            }
+        }
+    }
+
+    if (!foundHostName) {
+        await ext.ui.showWarningMessage(localize('unableFind', 'Unable to find "{0}" in SSH config.', host), { title: localize('addSSH', 'Add new SSH config host') });
```

Comment:
No real preference but it might be nice to show this for clarity (unless it looks cluttered as a button)
```suggestion
        await ext.ui.showWarningMessage(localize('unableFind', 'Unable to find "{0}" in SSH config.', host), { title: localize('addSSH', 'Add new SSH config host "{0}"', host) });
```

Reply:
![image](https://user-images.githubusercontent.com/5290572/104237801-6de58b00-540d-11eb-93f0-bb5767c0dc5f.png)

Reply:
That's like not even a button anymore 😂 My preference is keep the button simple since the name is already in the message

Reply:
I agree 😂 Too much button